File: packages/python/src/alumnium/tools/type_tool.py
from pydantic import Field

from alumnium.drivers.base_driver import BaseDriver

from .base_tool import BaseTool


class TypeTool(BaseTool):
    """Type text into an element. Automatically focuses the element and clears it before typing."""

    id: int = Field(description="Element identifier (ID)")
    text: str = Field(description="Text to type into an element")

    def invoke(self, driver: BaseDriver):
        driver.type(self.id, self.text)

File: packages/typescript/src/tools/TypeTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";
import { field, type FieldMetadata } from "./Field.ts";

export class TypeTool extends BaseTool {
  static description =
    "Type text into an element. Automatically focuses the element and clears it before typing.";
  static fields: FieldMetadata[] = [
    field({
      name: "id",
      type: "integer",
      description: "Element identifier (ID)",
    }),
    field({
      name: "text",
      type: "string",
      description: "Text to type into an element",
    }),
  ];

  id: number;
  text: string;

  constructor(args: { id: number; text: string }) {
    super();
    this.id = args.id;
    this.text = args.text;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.type(this.id, this.text);
  }
}
